src/ring_theory/roots_of_unity.lean

```diff
@@ -985,11 +933,10 @@ lemma squarefree_minpoly_mod {p : ℕ} [fact p.prime] (hdiv : ¬ p ∣ n) :
 /- Let `P` be the minimal polynomial of a root of unity `μ` and `Q` be the minimal polynomial of
 `μ ^ p`, where `p` is a prime that does not divide `n`. Then `P` divides `expand ℤ p Q`. -/
 lemma minpoly_dvd_expand {p : ℕ} (hprime : nat.prime p) (hdiv : ¬ p ∣ n) :
-  minpoly ℤ μ ∣
-  expand ℤ p (minpoly ℤ (μ ^ p)) :=
+  minpoly ℤ μ ∣ expand ℤ p (minpoly ℤ (μ ^ p)) :=
 begin
-  by_cases hn : n = 0, { simp * at *, },
-  have hpos := nat.pos_of_ne_zero hn,
+  rcases n.eq_zero_or_pos with rfl | hpos,
+  { simp * at *, },
```

Comment:
You don't use `hpos` in the second proof. Maybe you don't need the case split at all?

Reply:
it's I think a bit further down, you need it for `is_primitive_root.is_integral`

Reply:
It's used on L940 and 944, right?

Reply:
Ah right, I just assumed it wasn't because previously the assumption was called `hn`, and you didn't rename anything.